driver-core/src/main/java/com/datastax/driver/core/Metadata.java

```diff
@@ -633,20 +660,20 @@ private TokenMap(Token.Factory factory,
                 Host host = entry.getKey();
                 host.setTokens(ImmutableSet.copyOf(entry.getValue()));
             }
+            this.primaryToTokens = primaryToTokens;
         }
 
-        public static TokenMap build(Token.Factory factory, Map<Host, Collection<String>> allTokens, Collection<KeyspaceMetadata> keyspaces) {
+        public static TokenMap build(Token.Factory factory, Map<Host, ? extends Collection<Token>> allTokens, Collection<KeyspaceMetadata> keyspaces) {
 
             Set<Host> hosts = allTokens.keySet();
             Map<Token, Host> tokenToPrimary = new HashMap<Token, Host>();
             Map<Host, Set<Token>> primaryToTokens = new HashMap<Host, Set<Token>>();
```

Comment:
Now that the string -> token conversion is handled upstream, this is just a deep copy of `allTokens`.